x-pack/plugins/siem/public/components/page/hosts/hosts_table/index.tsx

```diff
@@ -68,54 +82,75 @@ const rowItems: ItemsPerRow[] = [
   },
 ];
 
-const Sup = styled.sup`
-  vertical-align: super;
-  padding: 0 5px;
-`;
-
 class HostsTableComponent extends React.PureComponent<HostsTableProps> {
+  private memoizedColumns: (
+    type: hostsModel.HostsType,
+    indexPattern: StaticIndexPattern
+  ) => Array<Columns<ValueOf<HostItem>>>;
+  private memoizedTitle: (totalCount: number) => JSX.Element;
+  private memoizedSorting: (
+    trigger: string,
+    sortField: HostsFields,
+    direction: Direction
```

Comment:
Since this is the pattern most tables will be following, we should probably extract this type and templatize it for use with any table. Not necessary now, but can be done as we work through the other tables to optimize them.